### PR TITLE
fix: flush indexed db on service-worker

### LINKED ISSUE #2237: Time to tackle cache more efficiently
It's a bit frustrating every time I need to clear the cache between new changes in deploy previews, even I see on `main` branch stuff which isn't there tho 😅 

Has someone will figure out the current caching issue to really flush stuff browser cache? 
In future, it would help to avoid unnecessary caching issues on the client browser while testing new stuff.

image

### LINKED ISSUE #3829: Can't load picture in gallery item detail, maybe cache issues
Not sure if it's related to `.avif` format
- #3828 

Weird is that it doesn't load only on beta, even after refresh 

https://beta.kodadot.xyz/rmrk/gallery/14086554-900D19DC7D3C444E4C-DOGSUN-DOG_SUNSET_22-0000000000000022
image

https://kodadot.xyz/rmrk/gallery/14086554-900D19DC7D3C444E4C-DOGSUN-DOG_SUNSET_22-0000000000000022
image

Incognito seems ok

image

Maybe some weird cache issues?

### PR DESCRIPTION
**Pull request by preschian**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).
👇 \_ Let's make a quick check before the contribution.

### PR type

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring

### What's new?

- [x] PR closes #3829
- [x] PR closes #2237
- [ ] Requires deployment <rubick/magick_issue>
- [ ] <brief_description_of_what_I've_added>

### Before submitting Pull Request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [ ] I've posted a screenshot of demonstrated change in this PR

### Optional

- [ ] I've tested it at </rmrk/collection/26902bc2f7c20c546a-1FVG7>
- [ ] I've tested PR on mobile and everything seems works
- [ ] I found edge cases
- [ ] I've written some unit tests 🧪

### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=DY4SQF2iD456tH89aQtz5wv1EV3BbSW8wKKuMcwbmXaj1pM)

### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

### Screenshot

- [ ] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

### CODE REVIEW

File: plugins/pwa.ts [hunk 1]
```diff
@@ -27,9 +38,15 @@ export default async () => {
         })
 
         notif.$on('close', () => {
-          window.sessionStorage.clear()
-          window.localStorage.clear()
-          window.location.reload()
+          try {
+            window.sessionStorage.clear()
+            window.localStorage.clear()
+            flushIndexedDb()
```
Comment (kkukelka) [inline plugins/pwa.ts:44]: i think we need an await here 
```suggestion
            await flushIndexedDb()
```

**Review by kkukelka — CHANGES_REQUESTED** [review]

**Review by kkukelka — APPROVED** [review]
lg now

**Review by roiLeo — COMMENTED** [review]
so, if I understand correctly, with each new version, the client indexedDB dabase will be deleted?

Can't we just clear kodadot indexedDB instead?

**Comment by preschian** [comment]
> so, if I understand correctly, with each new version, the client indexedDB dabase will be deleted?
> 
> Can't we just clear kodadot indexedDB instead?

afaik indexedDB does not share across domains (even subdomain), so this only deletes kodadot indexedDB

**Review by roiLeo — APPROVED** [review]
> afaik indexedDB does not share across domains (even subdomain), so this only deletes kodadot indexedDB

alright, noted!

**Comment by yangwao** [comment]
pay 50

**Comment by yangwao** [comment]
pay 50

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $50 @ 48.64 USD/KSM ~ 1.028 $KSM
🧗 [DY4SQF2iD456tH89aQtz5wv1EV3BbSW8wKKuMcwbmXaj1pM](https://kusama.subscan.io/account/DY4SQF2iD456tH89aQtz5wv1EV3BbSW8wKKuMcwbmXaj1pM)
🔗 [0x9e2ace1c51bebbd27c41ab65a7c7b9f7a20b28a4fd67718a74a63681cae90a4f](https://kusama.subscan.io/extrinsic/0x9e2ace1c51bebbd27c41ab65a7c7b9f7a20b28a4fd67718a74a63681cae90a4f)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**